File: 14.longest-common-prefix/main.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {

    pre = ""
    if (strs.length == 0) {
        return ""
    }
    minLen = strs[0].length
    for (i = 0; i < strs.length; i++) {
        if (strs[i].length < minLen) {
            minLen = strs[i].length
        }
    }
    for (i = 0; i < minLen; i++) {
        b = true
        for (j = 0; j < strs.length; j++) {
            if (strs[j][i] != strs[0][i]) {
                b = false
            }
        }
        if (b) {
            pre = pre + strs[0][i]
        } else {
            break
        }
    }
    return pre
};

(function (main) {
    console.log(longestCommonPrefix(["flower", "flow", "flight"]))
}())

File: 14.longest-common-prefix/main.go
package main

import "fmt"

func longestCommonPrefix(strs []string) string {

	pre := ""
	if len(strs) == 0 {
		return ""
	}

	for i := 0; i < minLen(strs); i++ {
		b := true
		for _, v := range strs {
			if v[i] != strs[0][i] {
				b = false
			}
		}
		if b {
			pre = pre + string(strs[0][i])
		} else {
			break
		}
	}
	return pre
}
func minLen(strs []string) int {
	l := len(strs[0])
	for _, v := range strs {
		if len(v) < l {
			l = len(v)
		}
	}
	return l
}
func main() {
	strs := []string{"flower", "flow", "flight"}
	fmt.Println(longestCommonPrefix(strs))
}
